spec/compiler/compiler_spec.cr

```diff
@@ -27,4 +27,15 @@ describe "Compiler" do
       end
     end
   end
+
+  it "treats all arguments post-filename as program arguments" do
+    with_tempfile "args_test" do |path|
+      `bin/crystal '#{compiler_datapath}/args_test.cr' -Dother_flag -- bar '#{path}'`
```

Comment:
I think referencing bin/crystal explicitly should be avoided if at all possible.
This would immediately fail on Windows, of course.

Maybe there should be a helper like this
https://github.com/crystal-lang/crystal/blob/127ea04edaee3e6eadd0715d85dc8b2bf8f2a955/spec/spec_helper.cr#L289 but I suppose this is explicitly about running Crystal directly, so maybe that one doesn't apply.

Also this is really bad in terms of shell quoting. Please use `Process.run` with explicit args.

*Also* needs `File.join`.